src/hoc/withHfpData.js

```diff
@@ -30,17 +52,28 @@ class HfpLoader extends React.Component {
       <HfpQuery route={route} date={date}>
         {({hfpPositions, loading}) => {
           if (loading || hfpPositions.length === 0) {
-            return children({hfpPositions: [], loading});
+            return children({
+              positionsByVehicle: [],
+              positionsByJourney: [],
+              loading,
+            });
           }
 
           const formattedPositions = formatData(hfpPositions);
           cacheData(formattedPositions, date, route);
 
-          return children({hfpPositions: formattedPositions, loading});
+          const positionsByVehicle = getGroupedByVehicle(formattedPositions);
+          const positionsByJourney = getGroupedByJourney(formattedPositions);
+
+          return children({positionsByVehicle, positionsByJourney, loading});
         }}
       </HfpQuery>
     ) : (
-      children({hfpPositions: cachedHfp, loading: false})
+      children({
+        positionsByVehicle: getGroupedByVehicle(cachedHfp),
+        positionsByJourney: getGroupedByJourney(cachedHfp),
```

Comment:
From now on, this component returns the hfp data in two versions. Grouped by vehicle or grouped by journey. Both are used in components below. This might need a localstorage reset, as the positions were cached in vehicle groups before.